raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -141,14 +143,33 @@ public KafkaRaftClient(NetworkChannel channel,
         this.retryBackoffMs = retryBackoffMs;
         this.electionTimeoutMs = electionTimeoutMs;
         this.electionJitterMs = electionJitterMs;
-        this.requestTimeoutMs = requestTimeoutMs;
         this.bootTimestamp = time.milliseconds();
         this.advertisedListener = advertisedListener;
         this.logger = logContext.logger(KafkaRaftClient.class);
         this.connections = new ConnectionCache(channel, bootstrapServers,
             retryBackoffMs, requestTimeoutMs, logContext);
         this.unsentAppends = new ArrayBlockingQueue<>(10);
         this.connections.maybeUpdate(quorum.localId, new HostInfo(advertisedListener, bootTimestamp));
+        this.recordAppender = records -> {
+            if (!quorum.isLeader()) {
```

Comment:
I'm neutral for a second state check inside append here. In the new API, the state machine should only accept appending logs after `becomeLeader` is invoked. But on the other hand, we couldn't enforce dropping of the appender during `becomeFollower` either, so it is still possible to see append requests from follower state machines.